src/shell-app.c

```diff
@@ -1428,6 +1420,24 @@ shell_app_create_custom_launcher_with_name (ShellApp *app,
 
   new_path = g_build_filename (g_get_user_data_dir (), "applications", shell_app_get_id (app), NULL);
 
+  if (!g_file_test (new_path, G_FILE_TEST_EXISTS))
+    {
+      /* Create a new 'X-Endless-CreatedBy' key to indicate the file was created
+       * by eos-desktop */
+      g_key_file_set_string (keyfile,
```

Comment:
I see your point. Looks fine then, since even if the desktop file would not be deleted in the case I mention, it would still be removed from the icon grid (which is what counts).